src/aspire/source/coordinates.py

```diff
@@ -0,0 +1,493 @@
+    """
+
+    def __init__(
+        self,
+        files,
+        particle_size,
+        max_rows=None,
+        dtype="double",
+    ):
+        """
+        :param files: A list of tuples of the form (path_to_mrc, path_to_coord)
+        :particle_size: Desired size of cropped particles
+        :param max_rows: Maximum number of particles to read. (If `None`, will
+        attempt to load all particles)
+        :param dtype: dtype with which to load images (default: double)
+        """
+        # get full filepaths and data folder
+        mrc_paths, coord_paths = self._check_and_get_paths(files)
+        # instantiate super
+        CoordinateSource.__init__(
+            self, mrc_paths, coord_paths, particle_size, max_rows, dtype
+        )
+
+    def coords_list_from_file(self, coord_file):
+        """
+        Given a coordinate file with (x,y) particle centers,
+        return a list of coordinates in our canonical (.box) format.
+        """
+        # check if it's a STAR file list of centers
+        if os.path.splitext(coord_file)[1] == ".star":
+            return self.coords_list_from_star(coord_file)
+        # otherwise we assume text file format with one coord per line:
+        with open(coord_file, "r") as infile:
+            lines = [line.split() for line in infile.readlines()]
+        return [self.box_coord_from_center(line, self.particle_size) for line in lines]
+
+
+class RelionCoordinateSource(CoordinateSource):
+    """
+    Represents a data source derived from an autopick.star file within a Relion
+    project directory.
+    """
+
+    def __init__(
+        self, relion_autopick_star, particle_size, max_rows=None, dtype="double"
+    ):
+        """
+                :param files: Relion STAR file e.g. autopick.star mapping micrographs to coordinate STAR files.
+                :particle_size: Desired size of cropped particles
+                :param max_rows: Maximum number of particles to read. (If `None`, will
+        attempt to load all particles)
+                :param dtype: dtype with which to load images (default: double)
+        """
+
+        # if not absolute path to star file, assume relative to working dir
+        if not os.path.isabs(relion_autopick_star):
+            relion_autopick_star = os.path.join(os.getcwd(), relion_autopick_star)
+
+        # the 'coordinate_files' block of the starfile specifies
+        # paths to micrographs and coordinate files relative to the
+        # Relion project dir
+        df = StarFile(relion_autopick_star)["coordinate_files"]
+        files = list(zip(df["_rlnMicrographName"], df["_rlnMicrographCoordinates"]))
+
+        # infer relion project dir since autopick.star will be at e.g.
+        # /relion/project/dir/Autopick/job00X/autopick.star
+        # get path 3 directories up
+        data_folder = Path(relion_autopick_star).parents[2]
```

Comment:
Hah, I've been inclined to ditch this entire class. 

The reason being that `CentersCoordinateSource` can read STAR coordinate files. So really you could just grab the list of coordinate files from `autopick.star` and give that to `CentersCoordinateSource`. 

This sketchy `Path` business is necessary if we want to find to the coordinate STAR files for the user, because they are referenced *relative* to the Relion project directory. I suppose we could pass in the Relion project dir as an argument, but you'd still have to give the path to `autopick.star` as well. Might be a confusing / not useful interface.

Reply:
Also easily confused with `RelionSource` since they're all in the `source` module

Reply:
I think this class is useful (and necessary). What concerns me is hardcoding the project path structure here. It's fine to use as a default, but it would be good to have an override option to supply another `data_folder`.